### PR TITLE
wishlist page

### PR DESCRIPTION
**Pull request by rohanmathur91**
## Description:
 - Added wishlist page
 - Implemented cart context for wishlist
 - Added route for wishlist page

## Link:
https://deploy-preview-5--evolt-ecommerce.netlify.app/

## Screenshot:
image
image

### CODE REVIEW

File: src/App.jsx [hunk 1]
```diff
@@ -10,6 +10,7 @@ function App() {
 			<Routes>
 				<Route path="/" element={<Home />} />
 				<Route path="/products" element={<Products />} />
+				<Route path="/Wishlist" element={<Wishlist />} />
```
Comment (Aditi0505) [inline src/App.jsx]: Suggestion: For consistency, you can start with lowercase in path. Is there a specific reason here for wishlist to be in title case? Let me know your thoughts about it?

**Review by Aditi0505 — APPROVED** [review]
Left one suggestion. Few things that I liked :
1. Good folder structure.
2. Good use of exporting all the related files from index.js
3. Clean coding standards.

I have an additional question:
What is the thought behind naming some folders in Pascal case while some in camelCase?

File: src/backend/db/products.js [hunk 2]
```diff
@@ -1,13 +1,10 @@
-import { v4 as uuid } from "uuid";
-
 /**
  * Product Database can be added here.
  * You can add products of your wish with different attributes
  * */
 
 export const products = [
 	{
-		_id: uuid(),
```
Comment (ishitakeshawani) [inline src/backend/db/products.js:10]: Why did you remove id from here?

**Review by ishitakeshawani — APPROVED** [review]
UI is nice and clean.
I liked that you have made two different providers.
It is responsive too.

File: src/App.jsx (hunk 1 above)
Comment (rohanmathur91) [reply src/App.jsx]: > Left one suggestion. Few things that I liked :
> 
> 1. Good folder structure.
> 2. Good use of exporting all the related files from index.js
> 3. Clean coding standards.
> 
> I have an additional question: What is the thought behind naming some folders in Pascal case while some in camelCase?

When naming a folder that consists of js functions we should name it in camelCase, while some folders consists of components, in that case, name it in Pascal case. Also not a hard and fast rule, we can follow any one convention but need to be consistent with that.

File: src/backend/db/products.js (hunk 2 above)
Comment (rohanmathur91) [reply src/backend/db/products.js:10]: Mockbee is internally generating an id, so for that I removed it from here, when you read the data i.e is GET request we will have an id in that.

File: src/App.jsx (hunk 1 above)
Comment (rohanmathur91) [reply src/App.jsx]: > Suggestion: For consistency, you can start with lowercase in path. Is there a specific reason here for wishlist to be in title case? Let me know your thoughts about it?

It's a typo, I will fix it.
